src/tools/pulse/generateInsightBrief/generatePulseInsightBriefTool.ts

```diff
@@ -0,0 +1,242 @@
+import { CallToolResult } from '@modelcontextprotocol/sdk/types.js';
+import { Err } from 'ts-results-es';
+
+import { getConfig } from '../../../config.js';
+import { useRestApi } from '../../../restApiInstance.js';
+import { PulseDisabledError } from '../../../sdks/tableau/methods/pulseMethods.js';
+import {
+  pulseInsightBriefRequestSchema,
+  PulseInsightBriefResponse,
+} from '../../../sdks/tableau/types/pulse.js';
+import { Server } from '../../../server.js';
+import { getTableauAuthInfo } from '../../../server/oauth/getTableauAuthInfo.js';
+import { Tool } from '../../tool.js';
+import { getPulseDisabledError } from '../getPulseDisabledError.js';
+
+const paramsSchema = {
+  briefRequest: pulseInsightBriefRequestSchema,
+};
+
+export type GeneratePulseInsightBriefError =
+  | {
+      type: 'feature-disabled';
+      reason: PulseDisabledError;
+    }
+  | {
+      type: 'datasource-not-allowed';
+      message: string;
+    };
```

Comment:
The `datasource-not-allowed` result is impossible, I think. That only comes from the tool scoping feature which the Pulse tools don't use.

Reply:
I guess that begs the question though... if a datasource filter is defined, should we first check the input metric uses that datasource?